src/sentry/issues/auto_source_code_config/derived_code_mappings_endpoint.py

```diff
@@ -0,0 +1,117 @@
+    InstallationNotFoundError,
+)
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+
+from .code_mapping import (
+    CodeMapping,
+    CodeMappingTreesHelper,
+    FrameInfo,
+    NeedsExtension,
+    create_code_mapping,
+)
+from .integration_utils import get_installation
+
+logger = logging.getLogger(__name__)
+
+
+def process_get_request(request: Request, organization: Organization) -> Response:
+    try:
+        file_repo_matches = []
+        resp_status: Literal[200, 204, 400] = status.HTTP_400_BAD_REQUEST
+
+        file_repo_matches = get_file_and_repo_matches(request, organization)
+        if file_repo_matches:
+            resp_status = status.HTTP_200_OK
+        else:
+            resp_status = status.HTTP_204_NO_CONTENT
+
+        return Response(serialize(file_repo_matches), status=resp_status)
+    except InstallationCannotGetTreesError:
+        return Response(
+            {"text": "The integration does not support getting trees"},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+    except InstallationNotFoundError:
+        return Response(
+            {"text": "Could not find this integration installed on your organization"},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+    except NeedsExtension:
+        return Response({"text": "Needs extension"}, status=status.HTTP_400_BAD_REQUEST)
+    except KeyError:
+        return Response({"text": "Missing required parameters"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+def get_file_and_repo_matches(request: Request, organization: Organization) -> list[dict[str, str]]:
+    frame_info = get_frame_info_from_request(request)
+    installation = get_installation(organization)
+    if not isinstance(installation, RepoTreesIntegration):
+        return []
+    trees = installation.get_trees_for_org()
+    trees_helper = CodeMappingTreesHelper(trees)
+    return trees_helper.get_file_and_repo_matches(frame_info)
+
+
+def get_frame_info_from_request(request: Request) -> FrameInfo:
+    frame = {
+        "absPath": request.GET.get("absPath"),
+        # Currently, the only required parameter
+        "filename": request.GET.get("stacktraceFilename"),
+        "module": request.GET.get("module"),
+    }
+    return FrameInfo(frame, request.GET.get("platform"))
+
+
+def process_post_request(request: Request, organization: Organization) -> Response:
+    try:
+        project = Project.objects.get(id=request.data.get("projectId"))
+        if not request.access.has_project_access(project):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+        installation = get_installation(organization)
+        # It helps with typing since org_integration can be None
+        if not installation.org_integration:
+            raise InstallationNotFoundError
+
+        code_mapping = get_code_mapping_from_request(request)
+        new_code_mapping = create_code_mapping(organization, code_mapping, project)
```

Comment:
Passing now `code_mapping` instead of the four parameters.
<img width="574" alt="image" src="https://github.com/user-attachments/assets/d6fd6165-0f32-4936-90ac-70738913d847" />